iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -166,13 +168,89 @@ void PortManager::handleReceiverPorts()
     }
 }
 
-void PortManager::handleInterfaces()
+void PortManager::handlePublisherPorts() noexcept
+{
+    // get the changes of publisher port offer state
+    for (auto publisherPortData : m_portPool->getPublisherPortDataList())
+    {
+        PublisherPortRouDiType publisherPort(publisherPortData);
+        auto maybeCaproMessage = publisherPort.getCaProMessage();
```

Comment:
have you thought about using this pattern
```
publisherPort.getCaProMessage(). and_then(...).or_else(...)
```

Reply:
Yes, I thought about it. Building blocks mostly use the `maybe` pattern. That's why I used it here as well. Is the `and_then(...)` pattern to way we want to go with iceoryx?

Reply:
if possible yes, since it makes the code in some situations cleaner

Reply:
Ok. I'm in!